content/docs/context.md

```diff
@@ -4,220 +4,92 @@ title: Context
 
 > Note:
->
-> As of React 16, `componentDidUpdate` no longer receives `prevContext`.
-
-## Referencing Context in Stateless Functional Components
-
-Stateless functional components are also able to reference `context` if `contextTypes` is defined as a property of the function. The following code shows a `Button` component written as a stateless functional component.
-
-```javascript
-import PropTypes from 'prop-types';
+> 
+> For more information about this pattern, see [render props](/docs/render-props.html).
 
-const Button = ({children}, context) =>
-  <button style={{background: context.color}}>
-    {children}
-  </button>;
+## Examples
 
-Button.contextTypes = {color: PropTypes.string};
-```
-
-## Updating Context
-
-Don't do it.
+### Static Context
 
-React has an API to update context, but it is fundamentally broken and you should not use it.
+Here is an example illustrating how you might inject a "theme" using context:
 
-The `getChildContext` function will be called when the state or props changes. In order to update data in the context, trigger a local state update with `this.setState`. This will trigger a new context and changes will be received by the children.
+`embed:context/theme-example.js`
 
-```javascript
-import PropTypes from 'prop-types';
+### Dynamic Context
 
-class MediaQuery extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {type:'desktop'};
-  }
+A more complex example with dynamic values for the theme:
 
-  getChildContext() {
-    return {type: this.state.type};
-  }
+**theme-context.js**
+`embed:context/theme-detailed-theme-context.js`
 
-  componentDidMount() {
-    const checkMediaQuery = () => {
-      const type = window.matchMedia("(min-width: 1025px)").matches ? 'desktop' : 'mobile';
-      if (type !== this.state.type) {
-        this.setState({type});
-      }
-    };
+**themed-button.js**
+`embed:context/theme-detailed-themed-button.js`
 
-    window.addEventListener('resize', checkMediaQuery);
-    checkMediaQuery();
-  }
+**app.js**
+`embed:context/theme-detailed-app.js`
 
-  render() {
-    return this.props.children;
-  }
-}
-
-MediaQuery.childContextTypes = {
-  type: PropTypes.string
-};
-```
+## Legacy API
 
-The problem is, if a context value provided by component changes, descendants that use that value won't update if an intermediate parent returns `false` from `shouldComponentUpdate`. This is totally out of control of the components using context, so there's basically no way to reliably update the context. [This blog post](https://medium.com/@mweststrate/how-to-safely-use-react-context-b7e343eff076) has a good explanation of why this is a problem and how you might get around it.
+> The legacy context API was deprecated in React 16.3 and will be removed in version 17.
```

Comment:
It wasn't deprecated. It is just legacy for now. We'll call it deprecated when we actually add deprecation warnings, and this will happen in a future minor release.